pkg/gofr/service/new.go

```diff
@@ -1,52 +1,96 @@
 package service
 
 import (
+	"bytes"
 	"context"
 	"fmt"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel/trace"
+	"gofr.dev/pkg/gofr/logging"
 	"net/http"
 	"net/http/httptrace"
 
 	"go.opentelemetry.io/contrib/instrumentation/net/http/httptrace/otelhttptrace"
-	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 	"go.opentelemetry.io/otel"
 )
 
 type httpService struct {
 	*http.Client
-
+	trace.Tracer
+	logging.Logger
 	url string
 }
 
-func (h *httpService) Get(ctx context.Context, path string, params map[string]interface{}) (*http.Response, error) {
-	uri := h.url + "/" + path
-
-	tr := otel.Tracer("gofr-http-client")
+type HTTP interface {
+	Get(ctx context.Context, api string, queryParams map[string]interface{}) (*http.Response, error)
 
-	ctx, span := tr.Start(ctx, uri)
-	defer span.End()
+	Patch(ctx context.Context, api string, queryParams map[string]interface{}, body []byte) (*http.Response, error)
+	PatchWithHeaders(ctx context.Context, api string, queryParams map[string]interface{}, body []byte, headers map[string]string) (*http.Response, error)
 
-	ctx = httptrace.WithClientTrace(ctx, otelhttptrace.NewClientTrace(ctx))
-	req, _ := http.NewRequestWithContext(ctx, "GET", uri, http.NoBody)
-	encodeQueryParameters(req, params)
-
-	return h.Do(req)
-}
+	Put(ctx context.Context, api string, queryParams map[string]interface{}, body []byte) (*http.Response, error)
+	PutWithHeaders(ctx context.Context, api string, queryParams map[string]interface{}, body []byte, headers map[string]string) (*http.Response, error)
 
-type HTTP interface {
-	Get(ctx context.Context, api string, params map[string]interface{}) (*http.Response, error)
+	Delete(ctx context.Context, api string, body []byte) (*http.Response, error)
+	DeleteWithHeaders(ctx context.Context, api string, body []byte, headers map[string]string) (*http.Response, error)
 }
 
-func NewHTTPService(serviceAddress string) HTTP {
+func NewHTTPService(serviceAddress string, logger logging.Logger) HTTP {
```

Comment:
This is breaking contract and the code example will not work with this. Also, because logger should be automatically taken care by the framework, one should not need to pass a new logger for using a service.